pkg/controller/compute-machineset/compute_machineset_controller.go

```diff
@@ -0,0 +1,519 @@
+	for _, machine := range allMachines {
+		if machine.DeletionTimestamp != nil {
+			continue
+		}
+		controllerRef := metav1.GetControllerOf(machine)
+		if controllerRef == nil {
+			c.logger.WithField("machineset", name).Debugf("no controller ref for machine %v", machine)
+			continue
+		}
+		if ms.UID != controllerRef.UID {
+			//c.logger.WithField("machineset", name).Debug("the kid is not my son %v", machine)
+			continue
+		}
+		filteredMachines = append(filteredMachines, machine)
+	}
+
+	var manageMachinesErr error
+
+	machineSetNeedsSync := c.expectations.SatisfiedExpectations(key)
+
+	if machineSetNeedsSync && ms.DeletionTimestamp == nil {
+		manageMachinesErr = c.manageMachines(filteredMachines, ms)
+	}
+
+	// update Status field
+	/***
+	original := ms
+	newms = ms.DeepCopy()
+	***/
+
+	if manageMachinesErr != nil {
+		return manageMachinesErr
+	}
+
+
+	return nil
+}
+
+func (c *Controller) manageMachines(machines []*clusteroperator.Machine, ms *clusteroperator.MachineSet) error {
+	machineSetKey, err := controller.KeyFunc(ms)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Couldn't get key for MachineSet %#v: %v", ms, err))
+		return nil
+	}
+
+	machinePrefix := ms.Name + "-"
+	//c.logger.WithField("machineset", ms.ObjectMeta.Name).Debugf("JDIAZ machine prefix: %v", machinePrefix)
+
+	machineSetMachines := []*clusteroperator.Machine{}
+	//machinesToCreate := []*clusteroperator.Machine{}
+	machinesToDelete := []string{}
+	//machinesToUpdate := []*clusteroperator.Machine{}
+
+	deleteCount := int(0)
+	createCount := int(0)
+
+
+	// caculate how many to create/delete
+	if len(machines) > ms.Spec.Size {
+		deleteCount = len(machines) - ms.Spec.Size
+	}
+	if len(machines) < ms.Spec.Size {
+		createCount = ms.Spec.Size - len(machines)
+	}
+
+	for _, machine := range machines {
+		/*
+		if currentMachineNeedsUpdate(machine, ms) {
+			c.logger.WithField("machineset", ms.Name).Errorf("JDIAZ child machine needs update %v", machine.Name)
+			machinesToDelete = append(machinesToDelete, machine)
+			createCount++
+			deleteCount++
+			continue
+		}
+		*/
+
+		machineSetMachines = append(machineSetMachines, machine)
+
+	}
+
+	for i := 0 ; i < deleteCount; i++ {
+		machinesToDelete = append(machinesToDelete, getMachineKey(machineSetMachines[i]))
+	}
+	c.logger.WithField("machineset", ms.Name).Infof("JDIAZ creating %v, deleting %+v", createCount, machinesToDelete)
+	c.expectations.SetExpectations(machineSetKey, createCount, machinesToDelete)
+
+	// Sync machines
+	machineConfig := buildNewMachine(ms, machinePrefix)
+	c.logger.WithField("machineset", ms.Name).Debugf("JDIAZ machine template: %+v", machineConfig)
+
+	for i := 0 ; i < createCount ; i++ {
+		_, err := c.client.ClusteroperatorV1alpha1().Machines(ms.ObjectMeta.Namespace).Create(machineConfig)
+		if err != nil {
+			// FIXME add to error list?
+			c.logger.WithField("machineset", ms.Name).Error("JDIAZ error creating machine...")
+			// update expectations?
```

Comment:
Update expectations by calling `c.expectations.CreationObserved(machineSetKey)` so that the controller is not waiting to observe the machine that the controller failed to create.